phoenix-core-client/src/main/java/org/apache/phoenix/util/CDCUtil.java

```diff
@@ -0,0 +1,118 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.phoenix.util;
+
+import java.sql.SQLException;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.NavigableSet;
+import java.util.Set;
+import java.util.StringTokenizer;
+
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.Scan;
+import org.apache.hadoop.util.StringUtils;
+import org.apache.phoenix.exception.SQLExceptionCode;
+import org.apache.phoenix.exception.SQLExceptionInfo;
+import org.apache.phoenix.execute.DescVarLengthFastByteComparisons;
+import org.apache.phoenix.schema.PTable;
+
+public class CDCUtil {
+    public static final String CDC_INDEX_PREFIX = "__CDC__";
+    public static final String CDC_INDEX_TYPE_LOCAL = "L";
+
+    /**
+     * Make a set of CDC change scope enums from the given string containing comma separated scope
+     * names.
+     *
+     * @param includeScopes Comma-separated scope names.
+     * @return the set of enums, which can be empty if the string is empty or has no valid names.
+     */
+    public static Set<PTable.CDCChangeScope> makeChangeScopeEnumsFromString(String includeScopes)
+            throws SQLException {
+        Set<PTable.CDCChangeScope> cdcChangeScopes = new HashSet<>();
+        if (includeScopes != null) {
+            StringTokenizer st  = new StringTokenizer(includeScopes, ",");
+            while (st.hasMoreTokens()) {
+                String tok = st.nextToken();
+                try {
+                    cdcChangeScopes.add(PTable.CDCChangeScope.valueOf(tok.trim().toUpperCase()));
+                }
+                catch (IllegalArgumentException e) {
+                    throw new SQLExceptionInfo.Builder(
+                            SQLExceptionCode.UNKNOWN_INCLUDE_CHANGE_SCOPE).setCdcChangeScope(
+                                    tok).build().buildException();
+                }
+            }
+        }
+        return cdcChangeScopes;
+    }
+
+    /**
+     * Make a string of comma-separated scope names from the specified set of enums.
+     *
+     * @param includeScopes Set of scope enums
+     * @return the comma-separated string of scopes, which can be an empty string in case the set is empty.
+     */
+    public static String makeChangeScopeStringFromEnums(Set<PTable.CDCChangeScope> includeScopes) {
+        String cdcChangeScopes = null;
+        if (includeScopes != null) {
+            Iterable<String> tmpStream = () -> includeScopes.stream().sorted()
+                    .map(s -> s.name()).iterator();
+            cdcChangeScopes = StringUtils.join(",", tmpStream);
+        }
+        return cdcChangeScopes;
+    }
+
+    public static String getCDCIndexName(String cdcName) {
+        return CDC_INDEX_PREFIX + SchemaUtil.getTableNameFromFullName(cdcName.toUpperCase());
+    }
+
+    public static boolean isCDCIndex(String indexName) {
```

Comment:
Seems like this API assumes that the indexName doesn't have the schema qualifier ?

Reply:
Yes, that is right. There are 2 calls to this right now, 1 is from the overloaded method that works on PTable and it is passing the name without schema qualifier. The other call comes from the "DROP INDEX" where you won't expect a schema qualifier anyway.